Src/PrimitiveUtilities.cpp

```diff
@@ -273,4 +273,167 @@ void Solver::createBox(const glm::vec3& translation, float scale, float k) {
+        uint32_t node10 = grid.gridIdToNodeId(currentNodeCount, {i + 1, j, 0});
+        uint32_t node11 = grid.gridIdToNodeId(currentNodeCount, {i + 1, j + 1, 0});
+
+        // Grid-aligned constraints
+        if (i < (grid.width - 1)) {
+            this->_distanceConstraints.push_back(createDistanceConstraint(
+                this->_constraintId++,
+                this->_nodes[node00],
+                this->_nodes[node10]));
+        }
+
+        if (j < (grid.height - 1)) {
+            this->_distanceConstraints.push_back(createDistanceConstraint(
+                this->_constraintId++,
+                this->_nodes[node00],
+                this->_nodes[node01]));
+        }
+
+        // Long diagonal constraints
+        if (i < (grid.width - 1) && j < (grid.height - 1)) {
+            this->_distanceConstraints.push_back(createDistanceConstraint(
+                this->_constraintId++,
+                this->_nodes[node00],
+                this->_nodes[node11]));
+        }
+    }
+  }
+
+  this->_triangles.reserve(
+      currentTriCount + (2 * grid.width * grid.height));
+  for (uint32_t i = 0; i < grid.width - 1; ++i) {
+    for (uint32_t j = 0; j < grid.height - 1; ++j) {
+      this->_triangles.push_back(
+          grid.gridIdToNodeId(currentNodeCount, {i, j, 0}));
+      this->_triangles.push_back(
+          grid.gridIdToNodeId(currentNodeCount, {i + 1, j, 0}));
+      this->_triangles.push_back(
+          grid.gridIdToNodeId(currentNodeCount, {i + 1, j + 1, 0}));
+
+      this->_triangles.push_back(
+          grid.gridIdToNodeId(currentNodeCount, {i, j, 0}));
+      this->_triangles.push_back(
+          grid.gridIdToNodeId(currentNodeCount, {i + 1, j + 1, 0}));
+      this->_triangles.push_back(
+          grid.gridIdToNodeId(currentNodeCount, {i, j + 1, 0}));
+    }
+  }
+
+  for (size_t i = currentDistConstraintsCount;
+       i < this->_distanceConstraints.size();
+       ++i) {
+    this->_distanceConstraints[i].setWeight(
+        1.0f - powf(1.0f - k, 1.0f / this->_options.iterations));
+  }
+
+  this->_lines.reserve(
+      currentLinesCount +
+      2 * (this->_distanceConstraints.size() - currentDistConstraintsCount));
+  for (size_t i = currentDistConstraintsCount;
+       i < this->_distanceConstraints.size();
+       ++i) {
+    const DistanceConstraint& constraint = this->_distanceConstraints[i];
+    this->_lines.push_back(constraint.getNodeId(0));
+    this->_lines.push_back(constraint.getNodeId(1));
+  }
+
+  this->_vertices.resize(this->_nodes.size());
+  for (size_t i = currentNodeCount; i < this->_nodes.size(); ++i) {
+    this->_vertices[i].position = this->_nodes[i].position;
+    this->_vertices[i].baseColor = boxColor;
+    this->_vertices[i].roughness = boxRoughness;
+    this->_vertices[i].metallic = boxMetallic;
+  }
+
+  this->renderStateDirty = true;
+}
+
+void Solver::createFloor(const glm::vec3& translation, float scale, float k) {
```

Comment:
Don't need the stiffness parameter `k` here.

Reply:
Fixed.